spec/src/main/asciidoc/java-api.asciidoc

```diff
@@ -129,7 +129,9 @@ public class SuccessfulCheck implements HealthCheck {
 
 The `name` is used to tell the different checks apart when a human operator looks at the responses.
 It may be that one check of several fails and it's useful to know which one.
-It's required that a response defines a name.
+It's required that a response defines a name, i.e. implementations must verify that a consistent name is provided when
+constructing a `HealthCheckResponse` instance, and fail accordingly if it is not.
+Specific tests to verify this behavior are executed when running the TCKs.
```

Comment:
Now that I'm reading this @fabiobrz, don't you think it would be better to fail the deployment when such scenario is detected? So rather than 503, test here for DeploymentException. Also please point me to where you have an implementation that passes these new tests ;)

Reply:
> Now that I'm reading this @fabiobrz, don't you think it would be better to fail the deployment when such scenario is detected? So rather than 503, test here for DeploymentException. Also please point me to where you have an implementation that passes these new tests ;)

Additionally, the new tests are checking something which happens when hitting the health endpoints, rather than when deploying.
And they only verify that a consistent name is provided when building responses via the static methods, rather than checking annotated health checks.